### ISSUE TITLE
forM_ and mapM_ and family require Monad, not Applicative

### ISSUE THREAD

**Issue by scshunt** [snoyberg/classy-prelude#108]
The functions `forM_` and `mapM_` (and other similar functions doing monadic folds) appearing in ClassyPrelude require that they be used for a `Monad`, when an `Applicative` would be enough. Is there a technical reason for this?

**Comment by snoyberg** [comment]
Yes: it would be a major breaking change on GHC 7.8, which we still support.

**Comment by gregwebs** [comment]
We could export a version with an A suffix

**Comment by scshunt** [comment]
It's not an inconvenience to me, I just noticed the inconsistency and had forgotten about the 7.8 thing.
